File: searching/tasks.py
import io,os, sys
from itertools import *
import math
from bisect import *

input = sys.stdin.readline
print = lambda x: sys.stdout.write(str(x))

alphs = "abcdefghijklmnopqrstuvwxyz"
digs = "0123456789"

INT_MIN = -1e10
INT_MAX = 1e10
MODVALUE = 10 ** 9 + 7

def gai():
    return list(map(int, input().strip().split()))

def test():
        n = int(input())
        tasks = []
        for i in range(n):
              task = gai()
              tasks.append(task)
        tasks.sort()
        time = 0
        res = 0
        for t in tasks:
            time += t[0]
            res += (t[1] - time)
        print(res)

test()

File: searching/tasks.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int n;
    cin >> n;
    vector<pair<int, int>> tasks(n);

    for (int i = 0; i < n; ++i) {
        cin >> tasks[i].first >> tasks[i].second;
    }

    // Sort tasks based on the first element of each pair
    sort(tasks.begin(), tasks.end());

    long long time = 0, res = 0;
    for (const auto& t : tasks) {
        time += t.first;
        res += (t.second - time);
    }

    cout << res << endl;
    return 0;
}
